lib/Dialect/Verif/VerifOps.cpp

```diff
@@ -42,6 +54,46 @@ OpFoldResult HasBeenResetOp::fold(FoldAdaptor adaptor) {
   return {};
 }
 
+//===----------------------------------------------------------------------===//
+// AssertLikeOps Canonicalizations
+//===----------------------------------------------------------------------===//
+
+namespace {
+struct AssertLikeOp {
+public:
+  // Pattern:
+  // assertlike(ltl.clock(prop, clk), en) -> clocked_assertlike(prop, en, clk)
+  template <typename TargetOp, typename Op>
+  static LogicalResult canonicalize(Op op, PatternRewriter &rewriter) {
+    // If the property is a block argument, then no canonicalization is possible
+    Value property = op.getProperty();
+    if (!property.getDefiningOp())
+      return success();
+
+    // Check for clock operand
+    if (auto clockOp = dyn_cast<ltl::ClockOp>(property.getDefiningOp()))
```

Comment:
nit: you can simplify this with getDefiningOp<..>. Also returning `success` is incorrect in the first condition.
```suggestion
    auto clockOp = op.getProperty.getDefiningOp<ltl::ClockOp>();
    if (!clockOp) return failure();
    ...
```

Reply:
that doesn't work apparently, like `getDefiningOp<ltl::ClockOp>()` doesn't compile anymore

Reply:
it's not invalid to have a block argument passed to an assertion, so the verifier shouldn't fail in that case, it should simply not have to check for a clock

Reply:
unless failing in a canonicalizer doesn't fail the entire compilation?

Reply:
Ah maybe I was missing `()`. Does ` op.getProperty().getDefiningOp<ltl::ClockOp>()` work?

Reply:
> it's not invalid to have a block argument passed to an assertion, so we shouldn't fail in that case, it should simply not have to check for a clock

`op.getProperty().getDefiningOp<ltl::ClockOp>()` is equivalent to `isa_and_nonnull<ltl::ClockOp>(op.getProperty().getDefiningOp())` so it works fine even when property is an argument

Reply:
no it doesn't